l10n_ar_account_reports/wizards/checks_to_date_report.py

```diff
@@ -0,0 +1,175 @@
+        # third_checks = self.env['account.payment']
+        # # _get_checks_to_date_on_state('holding', self.to_date, force_domain=force_domain).sorted(key=lambda r:r.check_id.payment_date or r.check_id.issue_date)
+
+        # # cheques propios entregados
+        # own_checks = self.env['account.payment']
+        # _get_checks_to_date_on_state('handed', self.to_date, force_domain=force_domain).sorted(key=lambda r:r.check_id.payment_date or r.check_id.issue_date)
+
+        # datadict = {
+        #     'third_checks': third_checks,
+        #     'own_checks': own_checks,
+        #     'date': self.to_date.strftime('%d/%m/%Y'),
+        #     'journal': self.journal_id.id
+        # }
+
+        # return self.env.ref('l10n_ar_account_reports.checks_to_date_report').report_action([], data=datadict)
+        return self.env.ref('l10n_ar_account_reports.checks_to_date_report').report_action(self)
+
+    @api.model
+    def _get_checks_handed(self, journal_id, to_date):
+        '''
+        hacemos una query que:
+        * toma todos los pagos correspondientes a un cheque
+        * dentro de esos pagos filtramos por los que no tengan matched_debit_ids
+        * dentro de account.full.reconcile buscar todas las que tengas asocido un pago que corresponda a un cheque, una vez filtrado,
+        * obtener la otra linea de ese full reconcile ya que esta corresponde a la conciliacion hecha en el date determinado
+        * entonces lo que filtramos finalmente sera, los cheques aun no fueron debitados y sean de fecha anterior a la dada o los que tiene debito luego de la fecha dada
+
+        -> en account.full.reconcile:
+        reconciled_line_ids.filtered(lambda x: x.move_id.payment_id.payment_method_id.code == 'check_printing').payment_id.ids -> nos da los id de los pagos correspondientes a un cheque que fueron conciliados
+        -> esto nos da en la fecha en la cual fueron debitados:
+        self.env['account.full.reconcile'].search([]).reconciled_line_ids.filtered(lambda x: x.move_id.payment_id.payment_method_id.code == 'check_printing')
+        .full_reconcile_id.reconciled_line_ids.filtered(lambda x: x.statement_line_id).mapped(lambda x: x.date)
+        '''
+        to_date = str(to_date)
+        query = """
+                SELECT DISTINCT ON (t.check_id) t.check_id AS cheque FROM
+                    (
+                        SELECT ap.id as check_id, ap_move.date as operation_date, apm.code as operation_code
+                        FROM account_payment ap
+                        LEFT JOIN account_payment_method AS apm ON apm.id = ap.payment_method_id
+                        LEFT JOIN account_move AS ap_move ON ap.move_id = ap_move.id
+                        LEFT JOIN account_journal AS journal ON ap_move.journal_id = journal.id
+                        WHERE
+                        apm.code = 'check_printing' AND ap_move.date <= '""" + to_date + """' order by ap.id, ap_move.date desc, ap.id desc
```

Comment:
you could also use 
```suggestion
                        apm.code = 'check_printing' AND ap_move.date <= $s order by ap.id, ap_move.date desc, ap.id desc
```

and then at the end of the SQL string add """ % (to_date)